tools/dds/dds-server/rs-dds-server.cpp

```diff
@@ -91,6 +92,127 @@ void start_streaming( std::shared_ptr< tools::lrs_device_controller > lrs_device
+                    static_cast< int16_t >( vsp.unique_id() ),
+                    static_cast< int8_t >( vsp.fps() ),
+                    vsp.format(),
+                    vsp.stream_type(),
+                    static_cast< int16_t >( vsp.width() ),
+                    static_cast< int16_t >( vsp.height() ) };
+
+
+            video_stream_profiles_msg.profiles_vec.push_back( std::move( vsp_msg ) );
+        }
+        else if( stream_profile.is< rs2::motion_stream_profile >() )
+        {
+            const auto & msp = stream_profile.as< rs2::video_stream_profile >();
+            device::notifications::motion_stream_profile msp_msg
+                = { static_cast< int8_t >( msp.stream_index() ),
+                    static_cast< int16_t >( msp.unique_id() ),
+                    static_cast< int8_t >( msp.fps() ),
+                    msp.format(),
+                    msp.stream_type() };
+            motion_stream_profiles_msg.profiles_vec.push_back(std::move(msp_msg));
+        }
+        else
+        {
+            LOG_ERROR( "got illegal profile with uid:" << stream_profile.unique_id() );
+        }
+    }
+}
+
+void publish_profiles( rs2::device dev, std::shared_ptr<librealsense::dds::dds_device_server> &server )
+{
+    using namespace librealsense::dds::topics;
+    auto stream_idx = 0;
+    device::notifications::video_stream_profiles video_stream_profiles_msg;
+    device::notifications::motion_stream_profiles motion_stream_profiles_msg;
+
+    // For each sensor publish all it's profiles
+    for( auto &sensor : dev.query_sensors() )
+    {
+        auto && stream_profiles = sensor.get_stream_profiles();
+        send_stream_header_msg( server, sensor, stream_idx++, stream_profiles.size() );
+
+        // Prepare stream profiles messages
+        prepare_profiles_messeges( dev,
+                                   stream_profiles,
+                                   video_stream_profiles_msg,
+                                   motion_stream_profiles_msg );
+
+        // Send video stream profiles
+        raw::device::notifications raw_video_stream_profiles_msg;
+        device::notifications::construct_raw_message(
+            device::notifications::msg_type::VIDEO_STREAM_PROFILES,
+            video_stream_profiles_msg,
+            raw_video_stream_profiles_msg );
+        server->publish_notifications( raw_video_stream_profiles_msg, true );
+
+        // Send motion stream profiles
+        raw::device::notifications raw_motion_stream_profiles_msg;
+        device::notifications::construct_raw_message(
+            device::notifications::msg_type::MOTION_STREAM_PROFILES,
+            motion_stream_profiles_msg,
+            raw_motion_stream_profiles_msg );
+        server->publish_notifications( raw_motion_stream_profiles_msg, true );
+
+        // Send pose stream profiles ? TODO
+    }
+}
+
+
+void publish_streams_and_profiles( rs2::device dev, std::shared_ptr<librealsense::dds::dds_device_server>& server )
```

Comment:
I think the `server` should be `const &`?

Reply:
I tried it, it won't work.
We use the server parameter for calling his `non-const` methods (they cannot be `const` because they change inner data)
The compiler won't allow a const parameter to call non const functions.
So passing a `shared_ptr` is a reasonable trade-off, it should be a small enough class to pass and not too much work is done.